java/ql/test/experimental/query-tests/security/CWE-522/InsecureLdapAuth.java

```diff
@@ -0,0 +1,92 @@
+import java.util.Hashtable;
+
+import javax.naming.Context;
+import javax.naming.directory.DirContext;
+import javax.naming.directory.InitialDirContext;
+import javax.naming.ldap.InitialLdapContext;
+
+public class InsecureLdapAuth {
+	// BAD - Test LDAP authentication in cleartext using `DirContext`.
+	public void testCleartextLdapAuth(String ldapUserName, String password) {
```

Comment:
Missing tests: the case where string literals are used instead of `Context.XYZ`

Reply:
Added a new test case with string literals